common-npm-packages/azure-arm-rest/azCliUtility.ts

```diff
@@ -130,37 +128,42 @@ export async function getFederatedToken(connectedServiceName: string): Promise<s
     return oidc_token;
 }
 
-function initOIDCToken(connection: WebApi, projectId: string, hub: string, planId: string, jobId: string, serviceConnectionId: string, retryCount: number, timeToWait: number): Q.Promise<string> {
-    var deferred = Q.defer<string>();
-    connection.getTaskApi().then(
-        (taskApi: ITaskApi) => {
-            taskApi.createOidcToken({}, projectId, hub, planId, jobId, serviceConnectionId).then(
-                (response: TaskHubOidcToken) => {
-                    if (response != null) {
-                        tl.debug('Got OIDC token');
-                        deferred.resolve(response.oidcToken);
-                    }
-                    else if (response.oidcToken == null) {
-                        if (retryCount < 3) {
-                            let waitedTime = timeToWait;
-                            retryCount += 1;
-                            setTimeout(() => {
-                                deferred.resolve(initOIDCToken(connection, projectId, hub, planId, jobId, serviceConnectionId, retryCount, waitedTime));
-                            }, waitedTime);
-                        }
-                        else {
-                            deferred.reject(tl.loc('CouldNotFetchAccessTokenforAAD'));
-                        }
-                    }
-                },
-                (error) => {
-                    deferred.reject(tl.loc('CouldNotFetchAccessTokenforAAD') + " " + error);
-                }
-            );
+export async function initOIDCToken(connection: WebApi, projectId: string, hub: string, planId: string, jobId: string, serviceConnectionId: string, retryCount: number = 0, timeToWait: number = 2000): Promise<string> {
+    const taskApi = await connection.getTaskApi();
+
+    let token: TaskHubOidcToken;
+
+    try {
+        token = await taskApi.createOidcToken({}, projectId, hub, planId, jobId, serviceConnectionId)
+    } catch (error) {
+        // Handle AggregateError if available, since the package uses Node10 types.
+        // Otherwise handle generic error
+        if (error.name === 'AggregateError' && error['errors'] !== undefined) {
+            for (const err of error.errors) {
+                tl.error(err);
```

Comment:
It may be hard to understand from the logs where the error is coming from
```suggestion
                tl.error("Error while trying to get OIDC token: " + err);
```

Reply:
Added to either condition when the error is an instance of AggregateError and the other type